parse_expr_data.R

```diff
@@ -1,77 +1,69 @@
 #!/usr/bin/env Rscript
 
-# Load optparse we need to check inputs
 suppressPackageStartupMessages(require(optparse))
-# Load common functions
 suppressPackageStartupMessages(require(workflowscriptscommon))
+suppressPackageStartupMessages(require(garnett))
 
 # create a CDS object from raw expression matrix. 
 # The CDS will then be used as input in further steps of the workflow
 
 # parse options 
 option_list = list(
     make_option(
-        c("-e", "--expression-matrix"),
+        c("-r", "--ref-10x-dir"),
         action = "store",
-        default = NA,
+        default = "reference_10x_dir",
         type = 'character',
-        help = "Numeric matrix of expression values; rows: genes, columns: cells.
-        See http://cole-trapnell-lab.github.io/monocle-release/docs/#getting-started-with-monocle
-        for explanation"
-    ), 
+        help = "10X-type directory with reference expression data"
+    ),
     make_option(
-        c("-p", "--phenotype-data"),
+        c("-q", "--query-10x-dir"),
         action = "store",
-        default = NA,
+        default = "query_10x_dir",
         type = 'character',
-        help = "Table of phenotype data; rows: cells, columns: cell attributes 
-                (such as cell type, culture condition, day captured, etc.)"
+        help = "10X-type directory with query expression data"
     ),
     make_option(
-        c("-f", "--feature-data"),
+        c("-c", "--ref-output-cds"),
         action = "store",
-        default = NA,
+        default = "ref_cds.rds",
         type = 'character',
-        help = "Table of gene features, rows:features (e.g. genes), columns:
-                gene attributes (such as biotype, gc content, etc.)"
+        help = "output file path for reference CDS object in .rds format"
     ),
     make_option(
-        c("-o", "--output-file"),
+        c("-d", "--query-output-cds"),
         action = "store",
-        default = NA,
+        default = "query_cds.rds",
         type = 'character',
-        help = "output file for CDS object in .rds format"
+        help = "output file path for query CDS object in .rds format"
     )
 )
 
-opt = wsc_parse_args(option_list, mandatory = c('expression_matrix', 
-                                                'phenotype_data',
-                                                'feature_data', 
-                                                'output_file'))
+opt = wsc_parse_args(option_list)
+input_dirs = c(opt$ref_10x_dir, opt$query_10x_dir)
+cds_names = c(opt$ref_output_cds, opt$query_output_cds)
+# process input directories
+for(idx in 1:length(input_dirs)){
+    input_dir = input_dirs[idx] 
+    if(! file.exists(input_dir)) stop(paste('File ', input_dir, 'does not exist'))
```

Comment:
Please separate checking of inputs from the rest of the logic. We should check everything is ready to go before proceeding with anything.